### DISCUSSION TITLE
How to trigger makepkg at low priority, without breaking Fish tab completion?

### DISCUSSION THREAD

**Discussion by nyanpasu64** [Morganamilo/paru#625] [Q&A]
I prefer to run `makepkg` at a low priority, so the compilation batch jobs don't slow down interactive usage of my computer or applications I run. Since paru doesn't support running makepkg at low priority, I've added a fish alias to run paru itself at a low priority:

```fish
alias paru="nice -n 19 paru --upgrademenu"
```

Unfortunately this breaks fish tab completions. Open a Fish prompt, type `paru -S bash` (or any other package really), and press Tab. The terminal hangs for several seconds then errors out.

```
nyanpasu64@ryzen ~> paru -S basherror: completion reached maximum recursion depth, possible cycle?
nyanpasu64@ryzen ~> paru -S bash
bash           (core)  bashish                  (AUR)  bash-command-timer-git           (AUR)  bash-git-prompt-git         (AUR)
bash.d          (AUR)  bashlint                 (AUR)  bash-common-bundle-dependencies  (AUR)  bash-it-git                 (AUR)
bash3           (AUR)  bashlint-git             (AUR)  bash-common-environment          (AUR)  bash-language-server  (community)
bash4           (AUR)  bashmarks-git            (AUR)  bash-common-parse-options        (AUR)  bash-pinyin-completion      (AUR)
…and 16 more rows
```

Pressing Ctrl+C while the terminal is hung generally gives the correct completion result.

## Debugging

The tab completion script is `/usr/share/fish/vendor_completions.d/paru.fish` installed by the `paru-bin` AUR package. The entire script runs once upon the first paru tab completion, and not at all for subsequent tab completions (even when completing different strings), regardless if the alias and hang are present or not. It seems that fish runs the script once, then uses the script's output to perform tab completions.

I'm not sure how to debug the tab completion. `fish -d 3` and `fish_trace=on fish` result in no output during the tab completion.

https://github.com/fish-shell/fish-shell/issues/7389 seems related, but appeared in fish 3.2.0, and I'm on 3.3.1 and I still get the issue.

Interestingly, a function works to set the priority without breaking tab completion:

```
function paru
  nice -n 19 paru --upgrademenu $argv;
end
```

## Path forward

Q: Is there an option to run paru and/or makepkg at low priority?

I tried creating `~/path/makepkg` (you can instead use `~/.local/bin/makepkg` which is more likely to work out of the box). This seems to successfully run all build processes at nice level 19 (I hope I got the script and escaping right):

```bash
#!/bin/bash
exec nice -n 19 /usr/bin/makepkg "$@"
```

Q: Should I report the alias bug to fish instead of here? Or is it "works as intended"?

**Comment by PremModhaOfficial** [comment]
the same happens with me even without the alias !
it worked fine with 'yay' as soon as i install paru the `yay -S <TAB>` completions also break and do the same thing as your case
